ipa-core/src/protocol/context/malicious.rs

```diff
@@ -21,6 +21,7 @@ use crate::{
             validator::{self, BatchValidator},
             Base, Context as ContextTrait, InstrumentedSequentialSharedRandomness,
             SpecialAccessToUpgradedContext, UpgradableContext, UpgradedContext,
+            UpgradedMaliciousContext,
```

Comment:
It seems weird to be using both `UpgradedMaliciousContext` and `Upgraded` in this module. Is there a reason (besides avoiding writing the `NotSharded` type parameter value) to do it this way?

Reply:
hmm, I think this is a mistake, let me try to fix it